### PR TITLE
Link tag updates

### PR DESCRIPTION
**Pull request by philschanely**
## Description

Adds a `--subtext` modifier to `sage-link` to allow for special styling in a "subtext" context.

### Screenshots

Screen Shot 2021-01-12 at 4 34 04 PM

### Testing

See new entry under Link element http://localhost:4000/pages/element/link

### CODE REVIEW

**Comment by QuintonJason** [comment]
@philschanely is it quick to whip up a story for this component or is it not applicable at the moment?

File: packages/sage-assets/lib/stylesheets/patterns/elements/_link.scss [hunk 1]
```diff
@@ -41,3 +41,24 @@
...
+
+  &:hover {
+    color: sage-color(charcoal, 400);
+    text-decoration: underline;
+  }
+
+  &:focus {
+    color: sage-color(charcoal, 400);
+    text-decoration: none;
+    box-shadow: 0 0 0 2px sage-color(charcoal, 400);
```
Comment (monkeypox8) [inline packages/sage-assets/lib/stylesheets/patterns/elements/_link.scss:61]: Could also set this to `currentColor` should we want to extend the style later

**Review by monkeypox8 — APPROVED** [review]
👍🏼 LGTM. Is this the only place where we're adding an underline as a link's default state?

File: packages/sage-assets/lib/stylesheets/patterns/elements/_link.scss (hunk 1 above)
Comment (philschanely) [reply packages/sage-assets/lib/stylesheets/patterns/elements/_link.scss:61]: Good call!

**Review by pixelflips — APPROVED** [review]
LGTM! 👍🏼

**Pull request merged**